src/PerformanceTests/Utils/AsyncCountdownEvent.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+public sealed class AsyncCountdownEvent
```

Comment:
There are quite a few variations of this class floating the internet. Would you care providing a source so we can see if there's an update, if we run into trouble or anything?